CRM/Price/BAO/LineItem.php

```diff
@@ -298,7 +298,15 @@ public static function getLineItems($entityId, $entity = 'participant', $isQuick
         'tax_amount' => $dao->tax_amount,
         'price_set_id' => $dao->price_set_id,
       );
-      $lineItems[$dao->id]['tax_rate'] = CRM_Price_BAO_LineItem::calculateTaxRate($lineItems[$dao->id]);
+      $taxRates = CRM_Core_PseudoConstant::getTaxRates();
+      if (isset($lineItems[$dao->id]['financial_type_id']) && array_key_exists($lineItems[$dao->id]['financial_type_id'], $taxRates)) {
+        // We are close to output/display here - so apply some rounding at output/display level - to not show Tax Rate in all 8 decimals
+        $lineItems[$dao->id]['tax_rate'] = round($taxRates[$lineItems[$dao->id]['financial_type_id']], 3);
+      }
+      else {
+        // There is no Tax Rate associated with this Financial Type
+        $lineItems[$dao->id]['tax_rate'] = FALSE;
+      }
```

Comment:
Not to a function called calculateTaxRate - part of the fundamental issue I addressed was that TaxRate should -not- be calculated. If should simply be retrieved. I can move this into a functional called retrieveTaxRate;